native/acter/src/api/stream.rs

```diff
@@ -998,121 +252,169 @@ impl TimelineStream {
-                let response = client.media().upload(&content_type, file_buf).await?;
-
-                let info = assign!(FileInfo::new(), {
-                    mimetype: Some(mimetype),
-                    size: size.and_then(UInt::new),
-                });
-                let mut file_content = FileMessageEventContent::plain(name, response.content_uri);
-                file_content.info = Some(Box::new(info));
-                let mut edited_content =
-                    RoomMessageEventContent::new(MessageType::File(file_content.clone()));
-                let replacement =
-                    Replacement::new(event_id.to_owned(), MessageType::File(file_content).into());
-                edited_content.relates_to = Some(Relation::Replacement(replacement));
-
-                timeline.send(edited_content.into()).await;
-                Ok(true)
-            })
-            .await?
-    }
-
-    pub async fn send_file_reply(
-        &self,
-        uri: String,
-        name: String,
-        mimetype: String,
-        size: Option<u64>,
-        event_id: String,
-    ) -> Result<bool> {
-        if !self.is_joined() {
-            bail!("Can't send reply as file to a room we are not in");
-        }
-        let room = self.room.clone();
-        let my_id = room
-            .client()
-            .user_id()
-            .context("User not found")?
-            .to_owned();
-        let timeline = self.timeline.clone();
-        let event_id = EventId::parse(event_id)?;
-        let client = self.room.client();
-
-        RUNTIME
-            .spawn(async move {
-                let member = room
-                    .get_member(&my_id)
-                    .await?
-                    .context("Couldn't find me among room members")?;
-                if !member.can_send_message(MessageLikeEventType::RoomMessage) {
-                    bail!("No permission to send message in this room");
-                }
-
-                let path = PathBuf::from(uri);
-                let mut file_buf = std::fs::read(path)?;
-
-                let content_type = mimetype.parse::<mime::Mime>()?;
-                let response = client.media().upload(&content_type, file_buf).await?;
-
-                let info = assign!(FileInfo::new(), {
-                    mimetype: Some(mimetype),
-                    size: size.and_then(UInt::new),
-                });
-                let mut file_content = FileMessageEventContent::plain(name, response.content_uri);
-                file_content.info = Some(Box::new(info));
-                let content =
-                    RoomMessageEventContentWithoutRelation::new(MessageType::File(file_content));
-                let Some(reply_item) = timeline.item_by_event_id(&event_id).await else {
-                    bail!("Not found which item would be replied to")
+                let edited_content = match *draft {
+                    MsgContentDraft::TextPlain { body } => {
```

Comment:
I wonder if we can't move this entire part in a `fn into_edited_content(self, event_id: EventId) -> RoomMessageEventContent` on the `MsgContentDraft` instead. Looks cleaner.

and would allow for these to have unit tests... which would be really great!

Reply:
Fixed by https://github.com/acterglobal/a3/pull/1175/commits/690f63f3fa9ab49db1527d1f0ff80e705cac65d0